gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -37,10 +32,10 @@ public RequestPreProcessor(
    * This is called once before processing the request.
    *
    * @param originalRequest The original request received
-   * @return The modified request with any additional filters depending on the scope config
+   * @return The modified request with unique selections and any additional filters depending on the
+   *     scope config
    */
-  public EntitiesRequest transformFilter(
-      EntitiesRequest originalRequest, EntitiesRequestContext context) {
+  public EntitiesRequest process(EntitiesRequest originalRequest, EntitiesRequestContext context) {
```

Comment:
Q: Any specific reason to rename the method as we have one more down transformFilter for TraceRequest?  Are we planning to rename it as well in future?

Reply:
The reason I renamed it is this wasn't just transforming the filter, it was also modifying other parts of the request (selections), so the name "transformFilter" felt misleading. Since the class is called RequestPreProcessor, "process" seemed an apt name, but open to other suggestions.

In the meantime, will rename the trace version too.

Reply:
Trace version renamed.